lib/canvas_cc/canvas_cc/models/assignment.rb

```diff
@@ -21,7 +21,8 @@ def resources
     def assignment_resource
       resource = CanvasCc::CanvasCC::Models::Resource.new
       resource.identifier = @identifier
-      resource.href = "#{resource.identifier}/assignment-#{CGI::escape(title.downcase.gsub(/\s/, '-'))}.html"
+      safe_slug = CGI::escape(title.downcase.gsub(/\s/, '-'))[0...235] # titles can include Unicode
```

Comment:
Because this truncates after escape, this can result in a malformed slug as demonstrated by the filename in test ending in `%`.

Example:
```ruby
# escaped string is at maximum length
pad = 226
str = "X" * pad + "龙"
esc = CGI::escape(str)
trc = esc[0...235]
p str[pad..]                # => "龙"
p esc[pad..]                # => "%E9%BE%99"
p trc[pad..]                # => "%E9%BE%99"
p CGI::unescape(esc)[pad..] # => "龙"
p CGI::unescape(trc)[pad..] # => "龙"

# escaped string is one char over maximum length and is truncated.
pad = 227
str = "X" * pad + "龙"
esc = CGI::escape(str)
trc = esc[0...235]
p str[pad..]                # => "龙"
p esc[pad..]                # => "%E9%BE%99" 
p trc[pad..]                # => "%E9%BE%9" 🫢
p CGI::unescape(esc)[pad..] # => "龙"
p CGI::unescape(trc)[pad..] # => "\xE9\xBE%9" 😓
```

Reply:
Sure, but that doesn't matter as it's never converted back to unicode, it's just intended to be unique.